Add Streaming API docs (#3284)
wso2/docs-apim: en/docs/design/create-api/create-streaming-api/create-a-websocket-streaming-api.md

# Create a WebSocket API

Follow the instructions below to create a WebSocket API using the basic flow.

### Design a WebSocket API

> 1.  Sign in to the WSO2 API Publisher `https://<hostname>:9443/publisher` (e.g., `https://localhost:9443/publisher`).

Reviewer: ```suggestion
1.  Sign in to the WSO2 API Manager (WSO2 API-M) Publisher Portal `https://<hostname>:9443/publisher` (e.g., `https://localhost:9443/publisher`).
```

2.  Go to **CREATE API** and Click **Design New Streaming API**.

    [![Design New Streaming API]({{base_path}}/assets/img/learn/design-api/streaming-api/design-new-streaming-api.png)]({{base_path}}/assets/img/learn/design-api/streaming-api/design-new-streaming-api.png)

    <html><div class="admonition note">
      <p class="admonition-title">Note</p>

>       <p>The <b>CREATE</b> button will only appear for a user who has the <code>creator</code> role permission.</p>

Reviewer: Same as [1].
[1] https://github.com/wso2/docs-apim/pull/3284/files#r589208308

                 <td class="confluenceTd">WebSocket</td>
             </tr>
             <tr>
                 <td colspan="2" class="confluenceTd">Context</td>
                 <td class="confluenceTd">
                     <div class="content-wrapper">
                         <p>
                             <code>/websocket</code>
                         </p>
                         <div>
                             <div class="confluence-information-macro-body">
                                 <p>The API context is used by the Gateway to identify the API. 
          Therefore, the API context must be unique. This context is the API's root context when invoking the API through 
          the Gateway.</p>
                             </div>
                             <div class="confluence-information-macro confluence-information-macro-tip">
                                 <span class="aui-icon aui-icon-small aui-iconfont-approve confluence-information-macro-icon"></span>
                                 <div class="confluence-information-macro-body">
                                     <p>You can define the API's version as a parameter of its context by 

>           adding the <code>{version}</code> into the context. For example, <code>{version}/websocket</code>. The API Manager 

Reviewer: ```suggestion
          adding the <code>{version}</code> into the context. For example, <code>{version}/websocket</code>. The WSO2 API-M
```

Reviewer: Do the changes where applicapable

                 <td colspan="1" class="confluenceTd">1.0.0</td>
             </tr>
             <tr>
                 <td colspan="2" class="confluenceTd">Protocol</td>
                 <td colspan="1" class="confluenceTd">
                     <p>WebSocket</p>
                 </td>
             </tr>
             <tr>
                 <td colspan="2" class="confluenceTd">Endpoint</td>
                 <td colspan="1" class="confluenceTd">
                    <p>
                    Use one of the following endpoints.
                    <ul>
                        <li>ws://echo.websocket.org:80</li>
                        <li>wss://echo.websocket.org:443</li>
                    </ul>
                 </td>
             </tr>
         </tbody>
     </table>
             
      [![Create WebSocket API Form]({{base_path}}/assets/img/learn/design-api/streaming-api/websocket/create-websocket-streaming-api-form.png)]({{base_path}}/assets/img/learn/design-api/streaming-api/websocket/create-websocket-streaming-api-form.png)

4.  Click **CREATE** or **CREATE & PUBLISH** to create the API.

     <html>
     <div class="admonition note">
     <p class="admonition-title">Note</p>

>      <p>The <b>CREATE & PUBLISH</b> option will only appear when a user who has <code>publisher</code> permission adds the details for the <b>Endpoint</b> field.</p>

Reviewer: Similar to [1].
[1]https://github.com/wso2/docs-apim/pull/3284/files#r589208308

     </div>
     </html>

    The overview page of the newly created API appears. 
    
    [![WebSocket API Overview Page]({{base_path}}/assets/img/learn/design-api/streaming-api/websocket/websocket-streaming-api-overview-page.png)]({{base_path}}/assets/img/learn/design-api/streaming-api/websocket/websocket-streaming-api-overview-page.png)

### Configure the Topics

> Topics of a SSE API are always **Subscribe and Publish**, where flow of events can be either from the server (backend) to the client, or from the client to the server. By default, the WebSocket API will have a topic with name `/*`.

Reviewer: ```suggestion
Topics of an SSE API are always **Subscribe and Publish**, where the flow of events can be either from the server (backend) to the client, or from the client to the server. By default, the WebSocket API will have a topic with the name `/*`.
```

> Topics of an SSE API are always **Subscribe and Publish**, where the flow of events can be either from the server (backend) to the client, or from the client to the server. By default, the WebSocket API will have a topic with the name `/*`.

Reviewer 2: this is websocket not SSE ?

Author: Addressed in https://github.com/wso2/docs-apim/pull/3284/commits/eb257b9312044b69fbcc8e4dc5cc16a20fada05c
